pyproject.toml

```diff
@@ -23,30 +23,59 @@ dynamic = ["version"]
 dependencies = [
     "xarray>=2024.10.0",
     "numpy>=2.0.0",
-    "packaging",
     "universal-pathlib",
     "numcodecs",
     "ujson",
+    "packaging",
 ]
 
 [project.optional-dependencies]
+# non-kerchunk readers
 hdf_reader = [
     "fsspec",
     "h5py",
     "hdf5plugin",
     "imagecodecs",
     "imagecodecs-numcodecs==2024.6.1",
-    "numcodecs"
+]
+# kerchunk-based readers
+hdf5_reader = [
+    "kerchunk>=0.25.0",
+    "h5py",
+]
+netcdf3_reader = [
+    "kerchunk>=0.25.0",
+    "scipy",
+]
+fits_reader = [
+    "kerchunk>=0.25.0",
+    "astropy",
+]
+# un-implemented readers
+# tiff_reader = [
+#     "kerchunk>=0.25.0",
+#     "tifffile",
+# ]
+# grib_reader = ["kerchunk>=0.25.0"]
+# zarr_reader = ["zarr==3.0.0b2"]
+all_readers = [
+    "virtualizarr[hdf]",
+    "virtualizarr[hdf5]",
+    "virtualizarr[netcdf3]",
+    "virtualizarr[fits]",
+#    "virtualizarr[tiff]",
+#    "virtualizarr[grib]",
+#    "virtualizarr[zarr]",
+]
+# writers
+icechunk = ["icechunk"]
+all = [
+    "virtualizarr[all_readers]",
+    "virtualizarr[icechunk]",
```

Comment:
Right now I don't think this can possibly work, because icechunk requires zarr-python v3 but kerchunk requires <3.

Reply:
not without truly living on the bleeding edge 😅 

https://github.com/developmentseed/virtualize-nex-gddp-cmip6/blob/f331421ee87adcfb42cb4a1904c05ec1214eb932/pyproject.toml#L24

Reply:
> Not sure we should change the public pip install to use some random branch...

sorry, I wasn't suggesting you should use that approach, but just was sharing for information about what is/isn't possible. I should've been more clear. 

> Aside: Is there a reason you are using virtualizarr main there? Would it be helpful to issue another release?

I wanted strict type hinting (https://github.com/zarr-developers/VirtualiZarr/pull/306)